docs/api.rst

```diff
@@ -0,0 +1,141 @@
++-------------------+--------+--------------------------------------------------------+
+|max_questions      |integer |default: 10                                             |
+|                   |        |                                                        |
+|                   |        |The maximum number of questions you want back.          |
++-------------------+--------+--------------------------------------------------------+
+
+
+Responses
+---------
+
+All response bodies are in JSON.
+
+HTTP 200: Success
+~~~~~~~~~~~~~~~~~
+
+With an HTTP 200, you'll get back a set of results in JSON.
+
+::
+
+   {
+       "documents": [
+           {
+               "title": ...               # title of kb article
+               "url": ...                 # url of kb article
+               "slug": ...                # slug of kb article
+               "summary": ...             # paragraph summary of kb article (plaintext)
+           }
+           ...
+       ],
+       "questions": [
+           {
+               "id": ...                  # integer id of the question
+               "answers": ...             # list of answer ids
+               "content": ...             # content of question (in html)
+               "created": ...             # datetime stamp in iso-8601 format
+               "creator": ...             # JSON object describing the creator
+               "involved": ...            # list of JSON objects describing everyone who
+                                            participated in the question
+               "is_archived": ...         # boolean for whether this question is archived
+               "is_locked": ...           # boolean for whether this question is locked
+               "is_solved": ...           # boolean for whether this question is solved
+               "is_spam": ...             # boolean for whether this question is spam
+               "is_taken": ...            # FIXME:
+               "last_answer": ...         # id for the last answer
+               "num_answers": ...         # total number of answers
+               "locale": ...              # the locale for the question
+               "metadata": ...            # metadata collected for the question
+               "tags": ...                # tags for the qeustion
+               "num_votes_past_week": ... # the number of votes in the last week
+               "num_votes": ...           # the total number of votes
+               "product": ...             # the product
+               "solution": ...            # id of answer marked as a solution if any
+               "taken_until": ...         # FIXME:
+               "taken_by": ...            # FIXME:
+               "title": ...               # title of the question
+               "topic": ...               # FIXME:
+               "updated_by": ...          # FIXME:
+               "updated": ...             # FIXME:
```

Comment:
I decided that the API shouldn't try and give only the minimal amount of data for "top level" results that aren't the children of another object. Instead most uses of an object give the full data available to the API. The only time the API sends reduced amounts of data is when an object is referenced from another objects as a foreign key, such as the `creator` field here. It is a lot of fields, yes, but I haven't seen any evidence that it is a problematic amount of fields. We chose to optimize for a few larger API responses instead of making smaller responses and possibly needing to make many requests to get the needed information.